charts/ggscout/examples/hashicorpvault-k8s/README.md

```diff
@@ -0,0 +1,117 @@
+# HashiCorp Vault with Kubernetes Authentication
+
+This example demonstrates how to configure ggscout to authenticate with HashiCorp Vault using Kubernetes authentication when running in a Kubernetes cluster.
+
+## Prerequisites
+
+1. HashiCorp Vault with Kubernetes auth method enabled
+2. Proper Vault policies and roles configured
+3. ggscout deployed in a Kubernetes cluster
+
+## Vault Configuration
+
+### 1. Enable Kubernetes Auth Method
+
+```bash
+# Enable Kubernetes auth method
+vault auth enable kubernetes
+```
+
+See HashiCorp Vault reference [documentation](https://developer.hashicorp.com/vault/docs/auth/kubernetes#configuration)
+
+### 2. Configure Kubernetes Auth Method
+
+```bash
+# Configure the Kubernetes auth method
+vault write auth/kubernetes/config \
+    token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443" \
+    kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+```
+
+### 3. Create Vault Policy
+
+```bash
+# Create policy for ggscout
+vault policy write ggscout-policy - <<EOF
+# Allow reading secrets from KV2 engine
+path "secret/data/*" {
+  capabilities = ["read", "list"]
+}
+
+path "secret/metadata/*" {
+  capabilities = ["read", "list"]
+}
+
+# Allow reading from KV1 engine (if used)
+path "secret/*" {
+  capabilities = ["read", "list"]
+}
+EOF
+```
+
+### 4. Create Vault Kubernetes Role
+
+```bash
+# Create Kubernetes auth role
+vault write auth/kubernetes/role/ggscout \
+    bound_service_account_names=ggscout-vault \
```

Comment:
I would use the default service_account name created by the helm chart instead: `ggscout`